cmake/modules/FindDeprecated.cmake

```
@@ -88,10 +88,12 @@ if("SPARSE" IN_LIST Deprecated_FIND_COMPONENTS)
   # This code was deprecated after Zephyr v3.2.0
   if(SPARSE)
     message(DEPRECATION
-        "Setting SPARSE=${SPARSE} is deprecated. "
+        "Setting SPARSE='${SPARSE}' is deprecated. "
         "Please set ZEPHYR_SCA_VARIANT to 'sparse'"
     )
-    set_ifndef(ZEPHYR_SCA_VARIANT sparse)
+    if(${SPARSE} STREQUAL "y")
```

Better add quotes to ensure that other cases of quoted / `;`-list arguments causing weird behavior.
```suggestion
    if("${SPARSE}" STREQUAL "y")
```

---

Done, thx!